collector/lib/CollectorConfig.h

```diff
@@ -75,7 +78,19 @@ class CollectorConfig {
   bool IsProcessesListeningOnPortsEnabled() const { return enable_processes_listening_on_ports_; }
   bool ImportUsers() const { return import_users_; }
   bool CollectConnectionStatus() const { return collect_connection_status_; }
-  bool EnableExternalIPs() const { return enable_external_ips_; }
+
+  // EnableExternalIPs will check for the existence
+  // of a runtime configuration, and defer to that value
+  // otherwise, we rely on the feature flag (env var)
+  bool EnableExternalIPs() const {
+    if (runtime_config_.has_value()) {
+      auto cfg = runtime_config_.value();
+      auto network_cfg = cfg.network_connection_config();
+      return network_cfg.enable_external_ips();
```

Comment:
In this case, `network_connection_config()` returns a reference, so it can't be null

Reply:
In that case, could we change line 88 slightly to make it clearer?
```cpp
const auto& network_cfg = cfg.network_connection_config();
```

Also, `auto` on its own will create a copy of the object, which we might want to avoid here. So we probably want to change line 87 as well:
```cpp
const auto& cfg = runtime_config_.value();
```

https://stackoverflow.com/questions/29859796/c-auto-vs-auto
https://en.cppreference.com/w/cpp/language/auto

Reply:
I didn't know that was a thing, thanks for the heads up!